### PR TITLE
Refactor if statement, introduce nodes for elif and else

### PR DESCRIPTION
**Pull request by konstin**
The old if layout couldn't differentiate between an else block with a single if statement and an elif statement. Additionally we getting rid of the recursion in favor of a single if struct with a vec of elif/else children. This is accompanied by a big refactoring in ruff which removes a bunch of TODOs and false negatives.

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #20** <a href="https://app.graphite.dev/github/pr/astral-sh/RustPython-Parser/20" target="_blank">Graphite</a> 
    * **PR #21** <a href="https://app.graphite.dev/github/pr/astral-sh/RustPython-Parser/21" target="_blank">Graphite</a> 
      * **PR #22** <a href="https://app.graphite.dev/github/pr/astral-sh/RustPython-Parser/22" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/RustPython-Parser/22?utm_source=stack-comment).

File: ast/src/generic.rs [hunk 1]
```diff
@@ -624,6 +624,18 @@ impl From<StmtIf> for Ast {
...
+pub struct ElifElseClause {
+    pub range: TextRange,
+    pub test: Option<Expr>,
+    pub body: Vec<Stmt>,
+}
+
+impl Node for ElifElseClause {
+    const NAME: &'static str = "ElifElse";
+    const FIELD_NAMES: &'static [&'static str] = &["test", "body"];
+}
```
Comment (MichaReiser) [inline ast/src/generic.rs:637]: Unrelated to your change (and this PR): I think we can remove this. It was added for the Pyo3 support which I believe we removed too?

File: parser/src/python.lalrpop [hunk 2]
```diff
@@ -759,25 +759,24 @@ ClassPattern: ast::Pattern = {
...
-            .end();
-        // handle elif:
-        for i in s2.into_iter().rev() {
-            let x = ast::Stmt::If(
-                ast::StmtIf { test: Box::new(i.1), body: i.2, orelse: last, range: (i.0..end_location).into() }
-            );
-            last = vec![x];
-        }
+            .map(|last| last.end())
+            .unwrap_or(body.last().unwrap().end());
```
Comment (MichaReiser) [inline parser/src/python.lalrpop:776]: May be worth moving this into a callback to not always execute it
```suggestion
            .unwrap_or_else(|| body.last().unwrap().end());
```

**Review by MichaReiser — APPROVED** [review]
Can you run the `parser` benchmarks locally to get an idea of the runtime impact of the change?

File: parser/src/python.lalrpop (hunk 2 above)
Comment (konstin) [reply parser/src/python.lalrpop:776]: that will be easier once suite statements make this free

**Pull request merged**